SG14/ring.h

```diff
@@ -108,7 +108,15 @@ namespace sg14
 		template <bool C>
 		bool operator==(const ring_iterator<Ring, C>& rhs) const noexcept;
 		template <bool C>
+		bool operator!=(const ring_iterator<Ring, C>& rhs) const noexcept;
+		template <bool C>
 		bool operator<(const ring_iterator<Ring, C>& rhs) const noexcept;
```

Comment:
I guess it's all semantics. In my view comparison operations make no sense because that doesn't match a ring. In practical terms usage may be different. I'll let the actual authors, who're more experienced with their use, puzzle it out with you.